torch/_export/serde/serialize.py

```diff
@@ -178,27 +178,68 @@ def serialize_metadata(node: torch.fx.Node) -> Dict[str, str]:
     ret = {}
     if stack_trace := node.meta.get("stack_trace"):
         ret["stack_trace"] = stack_trace
-    module_fqn = node.meta.get("module_fqn")
-    # Need an explicit None check instead of walrus operator, because
-    # module_fqn can be the empty string if the node belongs to the root.
-    # The walrus operator returns False on an empty string :(
-    if module_fqn is not None:
-        ret["module_fqn"] = module_fqn
-    # TODO(angelayi) add nn_module_stack and source_fn
+
+    if nn_module_stack := node.meta.get("nn_module_stack"):
+        # Serialize to "fx_node_name:(orig_ref,type_str)"
+        nn_module_list = [
+            f"{k}:({v[0]},{serialize_operator(v[1])})"
+            for k, v in nn_module_stack.items()
+        ]
+        ret["nn_module_stack"] = ";".join(nn_module_list)
```

Comment:
On the serializer side, I feel we should always save these metadata. If they're absent, then maybe we should error.

Reply:
hm, I'm not sure if we should error because some nodes (like assert) dont have meta. Maybe we can warning?

Reply:
Actually I thought ab it and we can't error -- because not every node has an nn_module_stack if it's not in a module.